flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/context/CepTimerService.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.cep.context;
+
+import org.apache.flink.annotation.Internal;
+
+/**
+ * Enables to provide time characteristic to {@link org.apache.flink.cep.nfa.NFA} for use in
+ * {@link org.apache.flink.cep.pattern.conditions.IterativeCondition}.
+ */
+@Internal
+public interface CepTimerService {
+
+	/**
+	 * Current processing time as returned from {@link org.apache.flink.streaming.api.TimerService}.
+	 */
+	long currentProcessingTime();
+
+	/**
+	 * If the {@link org.apache.flink.cep.operator.CepOperator} works in Processing time.
+	 */
+	boolean isProcessingTime();
```

Comment:
Why do we need the `isProcessingTime`? 
If this method is needed, then it feels like the `TimerService` is the wrong place to contain this information, from a class responsibilities standpoint.

Reply:
This is needed to handle `(Cep)TimerContext#timestamp` accordingly in NFA (which does not have access to the info if it works in processingTime).